Updated EXCEPT macro, install target, OS X compatibility

davedissian opened the pull request:
I've done a number of changes here:
- Added CMAKE_DEBUG_POSTFIX so debug builds have a "_d" appended to the end of the filename, matching Ogre.
- Enhanced the CMake script to honour BUILD_SHARED_LIBS rather than always build shared libs.
- Added a simple install target
- Building samples/tools are now controlled with CMake flags.
- Updated the EXCEPT macro to detect for the existence of OGRE_EXCEPT and use that if possible, keeping compatibility with Ogre 1.9 stable.
- Fixes for OS X

scrawl commented on CMakeLists.txt:
```diff
@@ -26,6 +29,12 @@ add_definitions(-DOGRE_PLUGIN_DIR_REL="${OGRE_PLUGIN_DIR_REL}")
-add_subdirectory(Samples/ParticleUniverseDemo)
-add_subdirectory(Tools/AtlasImageTool)
+
+if(PU_BUILD_SAMPLES)
```
Here you are using variables without having defined them first. Result: they don't show up in cmake-gui as they should.

davedissian commented:
Just addressed the issue, my bad, I normally just invoke CMake directly. Added another fix to make CMake build the library as a framework on OS X to match Ogre better.
